UI/common-settings.cpp

```diff
@@ -0,0 +1,183 @@
+#include "common-settings.hpp"
+
+#include "audio-encoders.hpp"
+
+bool IsAdvancedMode(config_t *config)
+{
+	const char *outputMode = config_get_string(config, "Output", "Mode");
+	return (strcmp(outputMode, "Advanced") == 0);
+}
+
+OBSData CommonSettings::GetDataFromJsonFile(const char *jsonFile)
+{
+	char fullPath[512];
+	obs_data_t *data = nullptr;
+
+	int ret = GetProfilePath(fullPath, sizeof(fullPath), jsonFile);
+	if (ret > 0) {
+		BPtr<char> jsonData = os_quick_read_utf8_file(fullPath);
+		if (!!jsonData) {
+			data = obs_data_create_from_json(jsonData);
+		}
+	}
+
+	if (!data)
+		data = obs_data_create();
+	OBSData dataRet(data);
+	obs_data_release(data);
+	return dataRet;
+}
+
+void CommonSettings::GetConfigFPS(config_t *config, uint32_t &num,
+				  uint32_t &den)
+{
+	uint32_t type = config_get_uint(config, videoSection, "FPSType");
+
+	if (type == 1) //"Integer"
+		GetFPSInteger(config, num, den);
+	else if (type == 2) //"Fraction"
+		GetFPSFraction(config, num, den);
+	else if (false) //"Nanoseconds", currently not implemented
+		GetFPSNanoseconds(config, num, den);
+	else
+		GetFPSCommon(config, num, den);
+}
+
+double CommonSettings::GetConfigFPSDouble(config_t *config)
+{
+	uint32_t num = 0;
+	uint32_t den = 0;
+	CommonSettings::GetConfigFPS(config, num, den);
+	return (double)num / (double)den;
+}
+
+void CommonSettings::GetFPSCommon(config_t *config, uint32_t &num,
+				  uint32_t &den)
+{
+	const char *val = config_get_string(config, videoSection, "FPSCommon");
+
+	if (strcmp(val, "10") == 0) {
```

Comment:
Better to have a local data table and a loop:

```
struct FPSData { char const* val; int num; int den; } ;
static const FPSData dataTable[] = {
    { "10", 10, 1 },
    { "20", 20, 1 },
// etc.
};

// defaults
num = 30;
den = 1;

for (FPSData const& datum : dataTable)
{
    if (strcmp(val, datum.val))
        continue;

    num = datum.num;
    den = datum.den;
    break;
}
```

Reply:
Can come back to this in another work stream. 
Also I think a for loop here breaks a simultaneous instruction optimization compilers usually make.